### DISCUSSION TITLE
Does WarpX Have a Laser envelope model?

### DISCUSSION THREAD

**Discussion by jiecaix** [BLAST-WarpX/warpx#5260] [Q&A] [answered]
I noticed that the WarpX documentation mentions 'ponderomotive guiding center (PGC) models.' Has this been implemented in WarpX, or is there a way to define it directly in the source code?

**Comment by RemiLehe** [comment, marked as answer]
Thanks for the question!
Unfortunately, WarpX currently does not have an envelope model.
